packages/agoric-cli/src/commands/inter.js

```diff
@@ -125,18 +196,89 @@ export const makeInterCommand = async (
       env.AGORIC_KEYRING_BACKEND,
     );
 
+  /** @type {typeof import('child_process').execFileSync} */
+  // @ts-expect-error execFileSync is overloaded
+  const execFileSync = (file, args, ...opts) => {
+    try {
+      return rawExec(file, args, ...opts);
+    } catch (err) {
+      throw new InvalidArgumentError(
+        `${err.message}: is ${file} in your $PATH?`,
+      );
+    }
+  };
+
+  /** @param {number} ms */
+  const delay = ms => new Promise(resolve => setTimeout(resolve, ms));
+  const show = (info, indent = false) =>
+    stdout.write(
+      `${JSON.stringify(info, bigintReplacer, indent ? 2 : undefined)}\n`,
+    );
+
   const rpcTools = async () => {
+    // XXX pass fetch to getNetworkConfig() explicitly
     const networkConfig = await getNetworkConfig(env);
     const { agoricNames, fromBoard, readLatestHead, vstorage } =
       await makeRpcUtils({ fetch }, networkConfig).catch(err => {
         throw new CommanderError(1, 'RPC_FAIL', err.message);
       });
+
+    /**
+     * @param {string} from
+     * @param {number|string} [minHeight]
+     */
+    const storedWalletState = async (from, minHeight = undefined) => {
+      const m = boardSlottingMarshaller(fromBoard.convertSlotToVal);
+
+      const history = await vstorage.readFully(
+        `published.wallet.${from}`,
+        minHeight,
+      );
+
+      /** @type {{ Invitation: Brand<'set'> }} */
+      // @ts-expect-error XXX how to narrow AssetKind to set?
+      const { Invitation } = agoricNames.brand;
+      const coalescer = makeWalletStateCoalescer(Invitation);
+      // update with oldest first
+      for (const txt of history.reverse()) {
+        const { body, slots } = JSON.parse(txt);
+        const record = m.unserialize({ body, slots });
+        coalescer.update(record);
+      }
+      const coalesced = coalescer.state;
+      harden(coalesced);
+      return coalesced;
+    };
+
+    /**
+     * Get OfferStatus by id, polling until available.
+     *
+     * @param {string} from
+     * @param {string|number} id
+     * @param {number|string} minHeight
+     */
+    const pollOffer = async (from, id, minHeight) => {
```

Comment:
please move storedWalletState` and `pollOffer` to lib/wallet. maybe have a `makeWalletUtils` there that itself calls `makeRpcUtils` so that this command can have simply,
```js
const tools = makeWalletUtils(({ fetch }, networkConfig);
```

This should be done before merge so that it's available for other commands to align with this one.

Reply:
then `makeWalletUtils` would call `makeRpcUtils`? That would build another `agoricNames` at the cost of N RPC round trips. This is where `makeQueryClient` and `makeBoardClient` would come in. Should I do that now?

Reply:
ah. I get it now. da61094c5